hq_superset/utils.py

```diff
@@ -139,29 +188,109 @@ def _ensure_schema_created(domain):
+        if domain_permissions[CAN_WRITE_PERMISSION]:
+            platform_roles_names.append(
+                'gamma',
+            )
+        elif domain_permissions[CAN_READ_PERMISSION]:
+            additional_roles.append(
+                self._get_user_domain_read_only_role(domain)
+            )
+
+        if platform_roles_names:
+            additional_roles.extend(
+                self._get_platform_roles(platform_roles_names)
+            )
+
+        return additional_roles
+
+    @staticmethod
+    def _get_domain_access(domain):
+        from .hq_url import user_domain_roles
+        from .hq_requests import HQRequest
+
+        hq_request = HQRequest(url=user_domain_roles(domain))
+        response = hq_request.get()
+
+        if response.status_code != 200:
+            return {}, []
+
+        response_data = response.json()
+        hq_permissions = response_data['permissions']
+        roles = response_data['roles'] or []
+
+        # Map between HQ and CCA
+        permissions = {
+            CAN_WRITE_PERMISSION: hq_permissions["can_edit"],
+            CAN_READ_PERMISSION: hq_permissions["can_view"],
+        }
+        return permissions, roles
+
+    @staticmethod
+    def _user_has_no_access(permissions: dict):
+        user_has_access = any([permissions[p] for p in permissions])
+        return not user_has_access
+
+    def _get_platform_roles(self, roles_names):
+        platform_roles = []
+        for role_name in roles_names:
+            superset_role_name = HQ_ROLE_NAME_MAPPING.get(role_name, role_name)
+            role = self.sm.find_role(superset_role_name)
+            if role:
+                platform_roles.append(role)
+        return platform_roles
+
+    def _get_user_domain_read_only_role(self, domain):
+        role = self._get_domain_user_role(domain, current_user)
+        self.sm.set_role_permissions(role, self._read_permissions_for_user)
+        return role
+
+    def _get_domain_user_role(self, domain, user):
+        role_name = self._domain_user_role_name(domain, user)
+        role = self.sm.find_role(role_name)
+        if not role:
+            return self.sm.add_role(role_name)
+        return role
+
+    @property
+    def _read_permissions_for_user(self):
+        return self._get_view_menu_permissions(
+            menu_permissions=READ_ONLY_MENU_PERMISSIONS
+        )
+
+    def _get_view_menu_permissions(self, menu_permissions):
+        """
+        This method returns combinations for all view menus and permissions
+        """
+        menus_permissions = []
+        for view_menu_name, permissions_names in menu_permissions.items():
+            menus_permissions.extend([
+                self.sm.add_permission_view_menu(permission_name, view_menu_name)
+                for permission_name in permissions_names
+            ])
+        return menus_permissions
+
+    @staticmethod
+    def _domain_user_role_name(domain, user):
+        return f"{domain}_user_{user.id}_read_only"
```

Comment:
Hey @Charl1996 

I wanted to check with you if its possible to add the HQ role UUID to the name?
While looking at the ticket to expire permissions after every hour, I realized we currently don't a way of linking HQ role to the CCA role, which could be useful.
If its not a big lift, would be good to do it now since we are releasing this now.

Reply:
I don't think it would be that much of a big lift. I'll look into it.

Reply:
Awesome, thank you!

Reply:
@mkangia I had a look into this and hit a snag...the Admin role is a virtual role and is never persisted in the database, so if the user is admin we don't have a role ID that we could use.

Maybe we should rethink how we track the role linkage. I'm going to hold off on this for now.

Reply:
>  if the user is admin we don't have a role ID that we could use.

so the id could be "admin"?

>  I'm going to hold off on this for now.

We don't need to block on this. Fine to go ahead without the role id if needed.